### PR TITLE
Separate Value from Critic

### PR DESCRIPTION
**Pull request by axlbrown**
Value is not a Critic.
Building a sequential model is factored out.

### CODE REVIEW

**Review by LiuYangMage — COMMENTED** [review]
I would suggest not double number of networks in critic. This is kind of wired from my point of view...
Maybe we should explicitly define 2 critics in GAC agent class, and pass the two critics into value.train()
See my comment on line 382.
Another option, is to fix num_critic = 2, and always return the minimum value when called. So we view it as a black box.

File: policies/gac/networks.py [hunk 1]
```diff
@@ -376,9 +379,7 @@ def train(self, transitions, action_sampler, actor, critic, K):
         Line 14 of Algorithm 2.
         Get the Q value of the states and action samples.
         """
-        Q1, Q2 = critic(
-            tf.concat([states, actions], -1)
-        )
+        Q1, Q2 = critic(states, actions)
```
Comment (LiuYangMage) [inline policies/gac/networks.py]: This line is pretty dangerous. I mean, it assumes that the critic returns values, which will only happen when num_critic = 2. So why we give the option num_critic = 1 then?

Comment (axlbrown) [reply policies/gac/networks.py]: This is the wired stuff from the other implementation I guess. We can definitely change it.

**Review by LiuYangMage — APPROVED** [review]
The critic and value class looks pretty good to me now.
I don't understand target_policy_q in GAC agent, but I think it's fine to merge and we will improve GACagent later.

**Comment by axlbrown** [comment]
target_policy_q is just min. We don’t need it. We already know it’s just min.

**Comment by LiuYangMage** [comment]
I manually resolved the conflict on Github. It's easy.

**Pull request merged**